client/src/components/mapping/FhirMappingPanel/FhirResourceTree/index.tsx

```diff
@@ -195,10 +195,22 @@ const FhirResourceTree = ({ onClickCallback }: Props) => {
       .map(buildNodeFromAttribute)
       .filter(Boolean) as TreeNode[];
 
+  const buildSlicedNode = (attribute: Attribute): TreeNode[] => {
+    const children = attribute.slices.map(buildNodeFromAttribute)
+
+    const copy = Attribute.from({ ...attribute, slices: [] });
+    const base = buildNodeFromAttribute(copy);
+
+    return [base, ...children].filter(Boolean) as TreeNode[];
+  };
+
   const buildNodeFromAttribute = (attribute: Attribute): TreeNode | null => {
     const node = createNode(attribute, []);
 
-    if (attribute.types.length > 1) {
+    if (attribute.slices.length > 0) {
+      // if the attribute has slices, we build a node with these different slices inside
+      node.childNodes = buildSlicedNode(attribute);
+    } else if (attribute.types.length > 1) {
       // if the attribute has multiple types, we have to create as many
       // children as there are types.
       node.childNodes = buildMultiTypeNode(attribute);
```

Comment:
Oh, that's right! And now, I got your point on the spreadType in the other PR
I'll see which one is the best to keep

Reply:
Hum, we don't always fill the choices because sometimes, in the structure def, only attribute[x] will be present and not attribute[x].field.
Are you ok if we first check if there are choices and if not, we check for the multi types case?

Reply:
I think we can't keep both (too confusing), can't we just populate choices automatically when there are multiple types on a choice attribute ? ie: when adding a value[x] attribute, we call attribute.addChoice automatically for each type ?

Reply:
The problem would be when we have value[x] and then a element with value[x].field in the structure definition... We don't want to add the choice automatically if it's specified after